libethcore/Transaction.cpp

```diff
@@ -127,8 +134,14 @@ void TransactionBase::streamRLP(RLPStream& _s, IncludeSignature _sig) const
 	else
 		_s << "";
 	_s << m_value << m_data;
+
 	if (_sig)
-		_s << (m_vrs.v + 27) << (u256)m_vrs.r << (u256)m_vrs.s;
+	{
+		if (m_chainID >= 1)
```

Comment:
Use `-4` as the default chain id to have Frontier compatibility. You should name it somewhere as `static const int frontierChainId = -4` and use instead of `-1` as default argument value. 

Please also use `chainId` name instead of `chainID`.